eth/catalyst/api.go

```diff
@@ -589,89 +589,84 @@ func (api *ConsensusAPI) heartbeat() {
-						if growth > 0 {
-							if left := new(big.Int).Sub(ttd, htd); left.IsUint64() {
-								eta = time.Duration(float64(left.Uint64())/growth) * time.Second
-							} else {
-								eta = time.Duration(new(big.Int).Div(left, big.NewInt(int64(growth))).Uint64()) * time.Second
-							}
-						}
-					}
-					var message string
-					if htd.Cmp(ttd) > 0 {
-						if lastTransitionUpdate.IsZero() {
-							message = "Merge already reached, but no beacon client seen. Please launch one to follow the chain!"
-						} else {
-							message = "Merge already reached, but previously seen beacon client is offline. Please ensure it is operational to follow the chain!"
-						}
-					} else {
-						if lastTransitionUpdate.IsZero() {
-							message = "Merge is configured, but no beacon client seen. Please ensure you have one available before the transition arrives!"
-						} else {
-							message = "Merge is configured, but previously seen beacon client is offline. Please ensure it is operational before the transition arrives!"
-						}
-					}
-					if eta == 0 {
-						log.Warn(message)
+					if lastTransitionUpdate.IsZero() {
+						log.Warn("Post-merge network, but no beacon client seen. Please launch one to follow the chain!")
 					} else {
-						log.Warn(message, "eta", common.PrettyAge(time.Now().Add(-eta))) // weird hack, but duration formatted doesn't handle days
+						log.Warn("Previously seen beacon client is offline. Please ensure it is operational to follow the chain!")
 					}
 					offlineLogged = time.Now()
 				}
 				continue
+			}
+			if time.Since(offlineLogged) > beaconUpdateWarnFrequency {
+				if lastForkchoiceUpdate.IsZero() && lastNewPayloadUpdate.IsZero() {
+					log.Warn("Beacon client online, but never received consensus updates. Please ensure your beacon client is operational to follow the chain!")
+				} else {
+					log.Warn("Beacon client online, but no consensus updates received in a while. Please fix your beacon client to follow the chain!")
+				}
+				offlineLogged = time.Now()
+			}
+			continue
+		}
+		if time.Since(lastTransitionUpdate) <= beaconUpdateExchangeTimeout {
+			offlineLogged = time.Time{}
+			continue
+		}
+		if time.Since(offlineLogged) > beaconUpdateWarnFrequency {
+			// Retrieve the last few blocks and make a rough estimate as
+			// to when the merge transition should happen
+			var (
+				chain = api.eth.BlockChain()
+				head  = chain.CurrentHeader()
+				htd   = chain.GetTd(head.Hash(), head.Number.Uint64())
+			)
+			if htd.Cmp(ttd) >= 0 {
+				if lastTransitionUpdate.IsZero() {
+					log.Warn("Merge already reached, but no beacon client seen. Please launch one to follow the chain!")
+				} else {
+					log.Warn("Merge already reached, but previously seen beacon client is offline. Please ensure it is operational to follow the chain!")
+				}
+				offlineLogged = time.Now()
+				continue
+			}
+			var eta time.Duration
+			if head.Number.Uint64() > 0 && htd.Cmp(ttd) < 0 {
```

Comment:
You don't need the `htd.Cmp(ttd) < 0` check any more, , the >= case was covered above